### PR TITLE
LPS-155334 Merge FDS top components and contexts

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-155334 Merge FDS top components and contexts](https://issues.liferay.com/browse/LPS-155334)
- [FDS technical improvements checklist](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2002681985/FDS+technical+improvements+checklist)

### What is the goal of this PR?

This is a small garden-trimming task:
- There is no need for two layers if top components, so we are merging `DataSet` and `FrontendDataSet`, into `FrontendDataSet`.
- There is no need for two top contexts, so we are merging `AppContext` and `DataSetContext`, into `FrontendDataSetContext`.

There are no changes in component behavior. 

### What does it look like?

No changes in UI.

### CODE REVIEW

**Comment by markocikos** [comment]
Force-pushed, fixing conflicts. No additional changes.

**Comment by markocikos** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/timeline/Timeline.js [hunk 1]
```diff
@@ -49,15 +49,15 @@ TimelineEntry.propTypes = {
...
+function Timeline({frontendDataSetContext, items}) {
 	return (
 		<ClayList className={classNames('mb-0', 'timeline')}>
 			{items.map((item, i) => (
 				<TimelineEntry
 					key={i}
 					{...item}
 					borderBottom={i !== items.length - 1}
-					dataSetContext={dataSetContext}
+					frontendDataSetContext={frontendDataSetContext}
```
Comment (beltranrengifo) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/timeline/Timeline.js:60]: This seems a bit weird 🤔 
The view variable is already wrapped in the context provider in `FrontendDataSet.js` line 744 🤷

**Review by beltranrengifo — APPROVED** [review]
Hi @markocikos 
Looks good to me, no issues loading the FDS in sample, module, Objects, Remote apps...
I left you just a minor question, not really related this PR but it took my attention.
Thxs

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/timeline/Timeline.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/timeline/Timeline.js:60]: Yeah, I very much agree. This _might_ make sense if the component is exported and used from some other module, but it doesn't look like that's the case. This is on the parts of code that don't currently work, views other than table, so it's impossible to test. We can refactor these as part of tasks to enable those views.

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**